### DISCUSSION TITLE
Any luck running CP/Net or MP/M on the IMSAI8008esp replica?

### DISCUSSION THREAD

**Discussion by kepello** [thehighnibble/thehighnibble.github.io#159] [IMSAI8080esp replica]
I have found DSK image files for MP/M -- but they don't start, just hang on boot, despite being labeled as bootable.[http://cpmarchives.classiccmp.org/cpm/mirrors/www.unix4fun.org/z80pack/index.html](link here).

I also found a CP/NET disk, which will start, but doesn't return to a prompt.  Instead it shows:

```
CP/NET 1.2 Loader
=================

BIOS         D200H  2E00H
BDOS         C400H  0E00H
SNIOS   SPR  C100H  0300H
NDOS    SPR  B500H  0C00H
TPA          0000H  B500H

CP/NET 1.2 loading complete.
CCP.SPR ?
```

**Comment by udo-munk** [comment]
The MP/M disks will only run on Z80pack cpmsim, same as CP/M also MP/M needs a customized loader, BIOS, just much more complicated. There is no MP/M implemented on the IMSAI 8080 machine.

If I remember right the CP/NET stuff uses Z80 instructions, so won't run on a 8080. Looks like what happened, just jumps off the rails somewhere.

**Comment by nbreeden2** [comment]
Udo, good to see you on the forum. :)
CP/M is customized to the specific systems hardware. The ports used, potential memory mapped I/O, how memory paging in hardware is implemented etc. are all specific to the hardware in the system and how it's strapped.  You can't expect any random CP/M disk image to just work on any system. On a PC the BIOS is customized by the OEM for their specific hardware; this allowed MSDOS to run as the details were hidden away in the BIOS by the OEM. For CP/M the end user was expected to customize the BIOS when needed. When you changed the hardware, you became the OEM. The CP/M user manual set includes one on customizing CP/M for your machine's specifics. As the S-100 world matured manufactures such as Cromemco and NorthStar defined more specific hardware configurations then provided CP/M customized to their hardware. This of course creates a "chicken and the egg" issue. If you can't boot CP/M how can you build a customized version of it? That's a subject for a different thread.

**Comment by udo-munk** [comment]
Also the CP/NET client requires a BIOS that matches the hardware. The disks have a BIOS for Z80pack cpmsim hardware, as far as I know a CP/NET BIOS for IMSAI hardware does not exist. You cannot take CP/NET adapted for hardware A and expect that it works on different hardware B.
